frontend/src/pages/profile/tokens.js

```diff
@@ -86,134 +47,102 @@ export class UserTokens extends React.Component {
           </ClipboardCopy>
         )},
         {title: token.expires},
-        {title: <Button variant="danger" onClick={() => this.onDeleteTokenClick(token)}>Delete</Button>}
+        {title: <Button variant="danger" onClick={() => {
+          setTokenToDelete(token);
+          setIsDeleteTokenOpen(true);
+        }}>Delete</Button>}
       ]
     };
-  }
-
-  updateUrl () {
-    let params = [];
-    params.push('page=' + this.state.page);
-    params.push('pageSize=' + this.state.pageSize);
-    this.props.navigate('/profile/tokens?' + params.join('&'));
-  }
-
-  setPage = (_event, pageNumber) => {
-    this.setState({page: pageNumber}, () => {
-      this.updateUrl();
-      this.getTokens();
-    });
-  };
-
-  setPageSize = (_event, perPage) => {
-    this.setState({pageSize: perPage}, () => {
-      this.updateUrl();
-      this.getTokens();
-    });
+    return(shutupEslint);
   };
 
   // TODO: useEffect on add and delete modal close
-  getTokens () {
+  useEffect(() => {
     // First, show a spinner
-    this.setState({rows: [getSpinnerRow(4)], isEmpty: false, isError: false});
-    let params = {
-      'pageSize': this.state.pageSize,
-      'page': this.state.page
-    };
-    HttpClient.get([Settings.serverUrl, 'user', 'token'], params)
+    HttpClient.get([Settings.serverUrl, 'user', 'token'], {page: page, pageSize: pageSize})
       .then(response => HttpClient.handleResponse(response))
-      .then(data => this.setState({
-        rows: data.tokens.map((token) => this.tokenToRow(token)),
-        page: data.pagination.page,
-        pageSize: data.pagination.pageSize,
-        totalItems: data.pagination.totalItems,
-        totalPages: data.pagination.totalPages,
-        isEmpty: data.pagination.totalItems === 0
-      }))
+      .then(data => {
+        if (data?.tokens) {
```

Comment:
If there are to tokens we will get here anyway because `data?.tokens` is true condition. We should probably have here `data?.tokens.length > 0`
Otherwise we will not see correct empty state when we have 0 tokens